pkg/logs/tail_test.go

```diff
@@ -0,0 +1,323 @@
+			{SequenceNum: 2, RequestStart: 1601645172164667, Stream: "stdout", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2"},
+			{SequenceNum: 3, RequestStart: 1601645172164667, Stream: "stdout", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3"},
+			{SequenceNum: 4, RequestStart: 1601645172164667, Stream: "stdout", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4"},
+			{SequenceNum: 5, RequestStart: 1601645172164667, Stream: "stderr", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5"},
+			{SequenceNum: 6, RequestStart: 1601645172164667, Stream: "stderr", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6 6"},
+			{SequenceNum: 7, RequestStart: 1601645172164667, Stream: "stdout", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7"},
+			{SequenceNum: 8, RequestStart: 1601645172164667, Stream: "stdout", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8"},
+			{SequenceNum: 9, RequestStart: 1601645172164667, Stream: "stderr", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9"},
+			{SequenceNum: 10, RequestStart: 1601645172164667, Stream: "stdout", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "10 10 10 10 10 10 10 10 10 10 10 10 10 10"},
+		},
+	}
+
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Errorf("parseResponseData mismatch (-want +got):\n%s", diff)
+	}
+}
+
+func TestFilterStream(t *testing.T) {
+	for i, test := range []struct {
+		stream string
+		logs   []Log
+		explen int
+	}{
+		{
+			stream: "stdout",
+			logs: []Log{
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+			},
+			explen: 5,
+		},
+		{
+			stream: "stderr",
+			logs: []Log{
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+			},
+			explen: 3,
+		},
+		{
+			logs: []Log{
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+			},
+			explen: 8,
+		},
+	} {
+		out := filterStream(test.stream, test.logs)
+		if len(out) != test.explen {
+			t.Errorf("#%d: exp: %d != got: %d", i, test.explen, len(out))
+		}
+	}
+}
+
+// TestGetLinks tests both getLinks() and getTimeFromLink()
```

Comment:
As mentioned above, it would be nice if the docstrings could be a bit more _descriptive_ of what they're trying to achieve. Otherwise I don't see much _value_ in the current description. I guess if I was to rephrase what I'm asking it would be: help me to understand the purpose of the test without me having to read the code.

Reply:
Fixed